### PR TITLE
Make Sardine dependency optional

### LINKED ISSUE #714: Could we make Sardine dependency optional?
This would mean that only people using/initialising `DataStoreService` would need to declare `sardine` dependency.
We have to introduce some sort of lazy initialization not to need it during GoodData creation.

### PR DESCRIPTION
**Pull request by jimirocks**
Resolves #714

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java [hunk 1]
```diff
@@ -235,6 +235,18 @@ protected GoodData(GoodDataEndpoint endpoint, Authentication authentication, Goo
         auditEventService = new AuditEventService(getRestTemplate(), accountService, settings);
         executeAfmService = new ExecuteAfmService(getRestTemplate(), settings);
         lcmService = new LcmService(getRestTemplate(), settings);
+
+        try {
+            Class.forName("com.github.sardine.Sardine", false, getClass().getClassLoader());
+            dataStoreService = new DataStoreService(getHttpClient(), getRestTemplate(), gdcService, endpoint.toUri());
+        } catch (ClassNotFoundException e) {
+            if (logger.isWarnEnabled()) {
+                logger.warn("Optional dependency Sardine not found - WebDAV related operations are not supported");
```
Comment (liry) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java]: What about to log only in info level? Warning is for me like something that I should fix, but this is gonna be intentional in most cases.

Comment (jimirocks) [reply gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java]: yeah that sounds reasonable

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by liry** [comment]
Can we somehow test it?

**Comment by jimirocks** [comment]
Already included in #813

**Pull request closed without merging**